### DISCUSSION TITLE
Open XML-based document file format *.mmx for saving or exporting all my map objects

### DISCUSSION THREAD

**Discussion by Zorco1** [freeplane/freeplane#767] [Ideas]
Like many programs such as other mind mapping solutions like Xmind (*.xmind)..., I do want to save my maps as *.mmx file (like Office docx, xlsx,, pptx ). The main advantage is to get all my map elements at the same spot and I get a file that I can easily share. A lot more easier than sharing a folder with many problems with wrong path images references after moving.

=> A zip file named *.mmx, with the mm file and all images inside. Below a *.xmind file unzipped.

<img width="517" alt="Xmind_structure" src="https://user-images.githubusercontent.com/81738678/197334038-e98d97a2-a117-4aaf-9ff8-85755e199782.PNG">

Thanks for considering this useful upgrade.

Below a xmind map to show how to use Freeplane. To see the structure, unzip + unzip *.xmind

[Utiliser FreePlane.zip](https://github.com/freeplane/freeplane/files/9844079/Utiliser.FreePlane.zip)

**Comment by nnako** [comment]
Hi @Zorco1 , this will not be a direct solution for your question but rather a pointer to something which had a similar problem in mind.

there are a couple of posts out there which mention a Freeplane addon "[pack and go](https://github.com/freeplane/addons)" developed by @vboerchers . maybe this would be a good starting point. the last step might be zipping the created folder into the `.mmx` file mentioned by you.

would this be a first step in your intended direction?

**Reply by Zorco1** [reply]
Hi @Zorco1, I thank you for your help. This is a good starting point.
I've tried the last addon 0.5 (unfortunably the last update was pushed 4 years ago) and on freeplane 1.10.5 it's not working at all so I can't test it.
I think that such a functionnality should be native in freeplane.

**Comment by nnako** [comment]
please see https://github.com/nnako/freeplane-packer for a first prototype solution.
